File: advanced-programming-methods/labs/toy-interpreter-java/src/model/Out.java
package model;

/**
 * The type Out.
 */
public class Out implements IList
{
    private int size = 0;
    private String[] queue;

    /**
     * Constructs a new Out.
     */
    public Out()
    {
        queue = new String[Constants.MAX_SIZE];
    }

    /**
     * {@inheritDoc}
     */
    public void add(String value)
    {
        queue[size++] = value;
    }

    /**
     * {@inheritDoc}
     */
    public String getAll()
    {
        StringBuilder b = new StringBuilder();

        for (int i = 0; i < size; i++)
        {
            b.append(queue[i]);
            b.append("\n");
        }

        return b.toString();
    }

    /**
     * {@inheritDoc}
     */
    public String getLast()
    {
        if (size == 0)
        {
            return "";
        }

        return queue[size - 1];
    }

    /**
     * {@inheritDoc}
     */
    public void clear()
    {
        queue = new String[Constants.MAX_SIZE];
        size = 0;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString()
    {
        String out = "";
        if (size == 0)
        {
            return out;
        }

        for (int i = 0; i < size - 1; i++)
        {
            out += queue[i] + ", ";
        }
        out += queue[size - 1];

        return out;
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/Out.cs
using System;
using System.Text;

namespace Model
{
    /// <summary>
    /// The type Out.
    /// </summary>
    public class Out : IList
    {
        private int size = 0;
        private string[] queue;

        /// <summary> Constructs a new Out. </summary>
        public Out()
        {
            queue = new string[Constants.MAX_SIZE];
        }

        /// <inheritdoc />
        public virtual void Add(string value)
        {
            queue[size++] = value;
        }

        /// <inheritdoc />
        public string All
        {
            get
            {
                StringBuilder b = new StringBuilder();
            
                for (int i = 0; i < size; i++)
                {
                    b.Append(queue[i]);
                    b.Append("\n");
                }

                return b.ToString();
            }
        }

        /// <inheritdoc />
        public string Last
        {
            get
            {
                if (size == 0)
                {
                    return "";
                }

                return queue[size - 1];
            }
        }

        /// <inheritdoc />
        public void Clear()
        {
            queue = new string[Constants.MAX_SIZE];
            size = 0;
        }
            
        public override string ToString()
        {
            string @out = "";
            if (size == 0)
            {
                return @out;
            }

            for (int i = 0; i < size - 1; i++)
            {
                @out += queue[i] + ", ";
            }
            @out += queue[size - 1];

            return @out;
        }
    }

}

